apps/openssl.c

```diff
@@ -112,6 +112,8 @@ static size_t internal_trace_cb(const char *buf, size_t cnt,
 
         tid = CRYPTO_THREAD_get_current_id();
         hex = OPENSSL_buf2hexstr((const unsigned char *)&tid, sizeof(tid));
+        if (hex == NULL)
+            return 0;
         BIO_snprintf(buffer, sizeof(buffer), "TRACE[%s]:%s: ",
                      hex, OSSL_trace_get_category_name(category));
         OPENSSL_free(hex);
```

Comment:
The only purpose for the hex value is to distinguish thread IDs in the trace output. So stopping trace output silently is confusing and hard to find. It would be better to send the output anyway, giving a visual indication of the failure:

```C
        BIO_snprintf(buffer, sizeof(buffer), "TRACE[%s]:%s: ",
                     hex != NULL ? hex : "<null ptr?!>", OSSL_trace_get_category_name(category));
```